Test-ScriptVersion doesn't work if proxy with auth is in place

lusassl-msft opened the pull request:
**Issue:**
`Test-ScriptVersion` doesn't work if proxy with auth is in place. Customers with such a configuration are getting the following return: `The remote server returned an error: (407) Proxy Authentication Required.`

**Reason:**
This is caused by the fact that we do not provide credentials for authentication by default.

**Fix:**
Make use of `[System.Net.CredentialCache]::DefaultNetworkCredentials` to use the network credentials of the current security context (https://docs.microsoft.com/en-us/dotnet/api/system.net.credentialcache.defaultnetworkcredentials?view=net-5.0). 

**Validation:**
~~Please do not merge yet - will be validated together with affected customers (especially the `([System.Net.WebRequest]::GetSystemWebproxy()).IsBypassed("https://github.com") -eq $false` condition needs to be validated).~~
Validated with customer and in lab.

dpaulson45 commented on Shared/Test-ScriptVersion.ps1:
```diff
@@ -90,6 +90,11 @@ function Test-ScriptVersion {
+        if (([System.Net.WebRequest]::GetSystemWebproxy()).IsBypassed("https://github.com") -eq $false) {
+            $webClient = New-Object System.Net.WebClient
+            $webClient.Headers.Add("User-Agent", "PowerShell")
+            $webClient.Proxy.Credentials = [System.Net.CredentialCache]::DefaultNetworkCredentials
```
We don't end up using `$webClient` when Invoking the web request. Does this do anything?

lusassl-msft replied:
Yes, it affects the current PS session. There is no need to set this anywhere else.

lusassl-msft commented:
Validated with customer. PowerShell `Invoke-WebRequest` relies on the proxy settings set via Internet Explorer. I've adjusted the check and added a new function to validate if a proxy has been set or not.

Just set a non-existing proxy server - `Invoke-WebRequest` failed as expected in this case (proxy does not exist):
![image](https://user-images.githubusercontent.com/40993616/128361186-f82fc98a-f183-48eb-929f-310af98c19e8.png)

Without proxy server:
![image](https://user-images.githubusercontent.com/40993616/128361310-d6cd69f0-abec-4fb7-88ba-f32e762432e3.png)

We then decide, based on the return, to use the network credential of the current security context or not for `System.Net.WebClient` class (only valid for the current PS session).
